lib/elixir/lib/calendar/date/range.ex

```diff
@@ -0,0 +1,90 @@
+defmodule Date.Range do
+  @moduledoc """
+  Defines a range of dates.
+
+  A range of dates represents a discrete number of dates where
+  the first and last values are dates with matching calendars.
+
+  Ranges of dates can be either increasing (`first <= last`) or
+  decreasing (`first > last`). They are also always inclusive.
+
+  A range of dates implements the `Enumerable` protocol, which means
+  functions in the `Enum` module can be used to work with
+  ranges:
+
+      iex> range = Date.range(~D[2001-01-01], ~D[2002-01-01])
+      iex> Enum.count(range)
+      366
+      iex> Enum.member?(range, ~D[2001-02-01])
+      true
+      iex> Enum.reduce(range, 0, fn(_date, acc) -> acc - 1 end)
+      -366
+
+  """
+
+  defstruct [:first, :last, :kind, :first_rata_die, :last_rata_die]
+
+  def new(%{calendar: fc} = first, %{calendar: lc} = last) when fc == lc do
+    %Date.Range{
+      first: first,
+      last: last,
+      first_rata_die: Date.to_rata_die(first),
+      last_rata_die: Date.to_rata_die(last),
+      kind: get_range_kind(first, last),
+    }
+  end
+  def new(_, _) do
+    raise ArgumentError,
+      "Both dates must have matching calendars"
+  end
+
+  defp get_range_kind(first, last) when first > last, do: :desc
+  defp get_range_kind(_first, _last), do: :asc
+
+  defimpl Enumerable, for: Date.Range do
+    def member?(range, date) do
```

Comment:
Tried to change this, but I'm getting the following on `iex`:
```elixir
iex(1)> ~D[2000-02-22] < ~D[2001-01-01]
false
iex(3)> ~D[2000-02-22] > ~D[2001-01-01]
true
```

Any idea on why this is happening? Keeping it with the `rata_die` works fine tho, since it's just comparing numbers.